### PR TITLE
Bump nixpkgs

### PR DESCRIPTION
**Pull request by smatting**
Motivated by https://github.com/wireapp/wire-server/pull/4220

## Checklist

 - [ ] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

**Comment by smatting** [comment]
Building of hoogle docs fails because of 
```
hoogle> Using Parsec parser
hoogle> Configuring hoogle-5.0.18.4...
hoogle> CallStack (from HasCallStack):
hoogle>   withMetadata, called at libraries/Cabal/Cabal/src/Distribution/Simple/Utils.hs:368:14 in Cabal-3.10.3.0:Distribution.Simple.Utils
hoogle> Error: Setup: Encountered missing or private dependencies:
hoogle> crypton-connection ==0.3.2
```

The application however builds

File: nix/patches/hoogle.patch
Comment (MangoIV) [inline nix/patches/hoogle.patch:1]: Please use fetchPatch if possible.

File: nix/wire-server.nix [hunk 1]
```diff
@@ -518,7 +518,7 @@ in
       pkgs.netcat
       pkgs.niv
       pkgs.haskellPackages.apply-refact
-      (pkgs.python310.withPackages
+      (pkgs.python3.withPackages
```
Comment (MangoIV) [inline nix/wire-server.nix:521]: Have you tried building the docs? I remember there was something broken.

File: nix/patches/hoogle.patch
Comment (MangoIV) [reply nix/patches/hoogle.patch:1]: Why does doJailBreak not work? Build type custom?

File: nix/wire-server.nix (hunk 1 above)
Comment (smatting) [reply nix/wire-server.nix:521]: this is not the python that is used by the docs, it's not used by anything really. mainly a convience if you want to write python scripts in hack/bin

File: nix/patches/hoogle.patch
Comment (smatting) [reply nix/patches/hoogle.patch:1]: ah right, that amounts to the same thing. done in https://github.com/wireapp/wire-server/pull/4273/commits/2ffa78b1a8553b29366ebb7c9110ae959b565da2

Comment (MangoIV) [reply nix/patches/hoogle.patch:1]: there's some pathological cases where it doesn't work, thought you'd run into this with hoogle, I wouldn't have been surprised :P

**Review by MangoIV — APPROVED** [review]
we're good, I think

**Pull request merged**